dev-libs/rocm-opencl-runtime/rocm-opencl-runtime-5.1.3.ebuild

```diff
@@ -0,0 +1,106 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit cmake edo flag-o-matic prefix
+
+DESCRIPTION="Radeon Open Compute OpenCL Compatible Runtime"
+HOMEPAGE="https://github.com/RadeonOpenCompute/ROCm-OpenCL-Runtime"
+SRC_URI="https://github.com/ROCm-Developer-Tools/ROCclr/archive/rocm-${PV}.tar.gz -> rocclr-${PV}.tar.gz
+	https://github.com/RadeonOpenCompute/ROCm-OpenCL-Runtime/archive/rocm-${PV}.tar.gz -> rocm-opencl-runtime-${PV}.tar.gz"
+
+LICENSE="Apache-2.0 MIT"
+SLOT="0/$(ver_cut 1-2)"
+IUSE="debug test"
+RESTRICT="!test? ( test )"
+KEYWORDS="~amd64"
+
+RDEPEND=">=dev-libs/rocr-runtime-${PV}
+	>=dev-libs/rocm-comgr-${PV}
+	>=dev-libs/rocm-device-libs-${PV}
+	>=virtual/opencl-3
+	media-libs/mesa"
+DEPEND="${RDEPEND}"
+BDEPEND=">=dev-util/rocm-cmake-${PV}
+	media-libs/glew
+	test? ( >=x11-apps/mesa-progs-8.5.0[X] )
+	"
+
+PATCHES=(
+	"${FILESDIR}/${PN}-5.1.3-remove-clinfo.patch"
+	"${FILESDIR}/${PN}-3.5.0-do-not-install-libopencl.patch"
+)
+
+S="${WORKDIR}/ROCm-OpenCL-Runtime-rocm-${PV}"
+S1="${WORKDIR}/ROCclr-rocm-${PV}"
+
+src_prepare() {
+	# Remove "clinfo" - use "dev-util/clinfo" instead
+	[ -d tools/clinfo ] && rm -rf tools/clinfo || die
+
+	cmake_src_prepare
+
+	hprefixify amdocl/CMakeLists.txt
+
+	sed -e "s/DESTINATION lib/DESTINATION ${CMAKE_INSTALL_LIBDIR}/g" -i packaging/CMakeLists.txt || die
+	# remove trailing CR or it won't work
+	sed -e "s/\r$//g" -i tests/ocltst/module/perf/oclperf.exclude || die
+
+	pushd ${S1} || die
+	# Bug #753377
+	# patch re-enables accidentally disabled gfx800 family
+	eapply "${FILESDIR}/${PN}-5.0.2-enable-gfx800.patch"
+	popd || die
+}
+
+src_configure() {
+	# Reported upstream: https://github.com/RadeonOpenCompute/ROCm-OpenCL-Runtime/issues/120
+	append-cflags -fcommon
+
+	local mycmakeargs=(
+		-Wno-dev
+		-DROCCLR_PATH="${S1}"
+		-DAMD_OPENCL_PATH="${S}"
+		-DROCM_PATH="${EPREFIX}/usr"
+		-DBUILD_TESTS=$(usex test ON OFF)
+		-DEMU_ENV=ON
+		# -DCMAKE_STRIP=""
+	)
+	cmake_src_configure
+}
+
+src_install() {
+	insinto /etc/OpenCL/vendors
+	doins config/amdocl64.icd
+
+	cd "${BUILD_DIR}" || die
+	insinto /usr/lib64
+	doins amdocl/libamdocl64.so
+	doins tools/cltrace/libcltrace.so
+}
+
+check_rw_permission() {
+	[[ -r "$1" ]] && [[ -w "$1" ]] || die \
+		"${PORTAGE_USERNAME} do not have read or write permissions on $1! \n Make sure ${PORTAGE_USERNAME} is in render group and check the permissions."
```

Comment:
pkgcheck complains about this variable. Can we avoid using it and maybe use the same warning string as in `rocm.eclass`

EbuildReservedName: version 5.1.3: line 84: variable name "PORTAGE_USERNAME" is disallowed because "portage"is a reserved substring

Reply:
Seems that I force pushed the wrong version. Just ignore this and I'll fix up

Reply:
I don't recall what the conclusion for rocm.eclass were but you can use `${USER}` here, I think.

Reply:
> I don't recall what the conclusion for rocm.eclass were but you can use `${USER}` here, I think.

rocm.eclass uses "portage user":

```
		if [[ ! -r ${device} || ! -w ${device} ]]; then
			eerror "Cannot read or write ${device}!"
			eerror "Make sure it is present and check the permission."
			ewarn "By default render group have access to it. Check if portage user is in render group."
			die "${device} inaccessible"
		fi
```

Reply:
@mgorny I tried `${USER}` in `rocm.eclass` but it is an empty value, so I just switch it back to `portage user`